lib/difi_source_cpp_impl.h

```diff
@@ -52,19 +46,19 @@ class difi_source_cpp_impl : public difi_source_cpp<T>
 
     struct context_packet {
         u_int64_t class_id;
-        u_int32_t full;
-        u_int64_t frac;
+        u_int32_t full; //seconds
+        u_int64_t frac; //picoseconds
         u_int32_t cif;
         u_int32_t ref_point;
-        double bw;
-        u_int64_t if_ref_freq;
-        u_int64_t rf_ref_freq;
-        u_int64_t if_band_offset;
-        u_int32_t ref_lvl;
-        u_int32_t gain;
-        double samp_rate;
-        u_int64_t t_adj;
-        u_int32_t t_cal;
+        double bw; //Hertz
+        double if_ref_freq; //Hertz
+        double rf_ref_freq; //Hertz
+        double if_band_offset; //Hertz
+        float ref_lvl; //dBm
+        float rf_gain, if_gain; //dB
+        double samp_rate; //Hertz
+        int64_t t_adj; //femtoseconds
```

Comment:
is this really in femto seconds? not pico?

Reply:
Per the V49 spec `9.7 Temporal Fields`:

<em>"A description of a timing data type with a femtosecond resolution is provided, which is used in most of the timing fields in this section." </em>

So femtosecond is the default.

Reply:
Thanks, just wanted to double check since the fractional time portion of DIFI timestamps is pico
![image](https://user-images.githubusercontent.com/5722532/197249248-d859a3d0-9bbb-448b-be29-85a83e0840ed.png)

Reply:
About the timestamp adjustment field, the DIFI standard
regrettably says, "This defaults to 0 picoseconds,"
but refers to VITA 49.2 section 9.7.3.1 "for complete details."
VITA 49.2 rules 9.7.3.1-2  and 9.7-2 specify femtoseconds.